quick-dsp/src/bell202/mod.rs

```diff
@@ -93,11 +94,65 @@ where
         .apply_one_to_one(Decimator::<f32, Out>::default())
 }
 
+pub struct Ax25Debug<'a>(pub &'a [u8]);
+
+impl<'a> Ax25Debug<'a> {
+    /// Returns true if this looks like a AX25 packet
+    pub fn is_ax25(&self) -> bool {
+        let mut addr_len = 0u32;
+        for x in self.0.iter() {
```

Comment:
Could you do, `for &x in self.0.iter() {` and elide the `let` statement on the next line?

This could also possibly be,

    `let addr_len = self.0.iter().take_while(|&x| x & 1 != 1).count() + 1;`

Reply:
A little harder to read, but I like it better. I'll refactor this code.